i gave this a test, with this config
```yaml
sensor:
  - platform: opensky
    radius: 10   # kilometers
```

and got this error
```python
Traceback (most recent call last):
  File "/usr/lib/python3.5/asyncio/tasks.py", line 241, in _step
    result = coro.throw(exc)
  File "/usr/local/lib/python3.5/dist-packages/homeassistant/helpers/entity_component.py", line 359, in async_process_entity
    new_entity, self, update_before_add=update_before_add
  File "/usr/local/lib/python3.5/dist-packages/homeassistant/helpers/entity_component.py", line 189, in async_add_entity
    yield from self.hass.loop.run_in_executor(None, entity.update)
  File "/usr/lib/python3.5/asyncio/futures.py", line 361, in __iter__
    yield self  # This tells Task to wait for completion.
  File "/usr/lib/python3.5/asyncio/tasks.py", line 296, in _wakeup
    future.result()
  File "/usr/lib/python3.5/asyncio/futures.py", line 274, in result
    raise self._exception
  File "/usr/lib/python3.5/concurrent/futures/thread.py", line 55, in run
    result = self.fn(*self.args, **self.kwargs)
  File "/home/brenda/.homeassistant/custom_components/sensor/opensky.py", line 109, in update
    if distance > self._radius:
TypeError: unorderable types: NoneType() > float()
```

--- a/homeassistant/components/sensor/opensky.py
+++ b/homeassistant/components/sensor/opensky.py
@@ -13,9 +13,11 @@ import voluptuous as vol
 from homeassistant.components.sensor import PLATFORM_SCHEMA
 from homeassistant.const import (
     CONF_NAME, CONF_LATITUDE, CONF_LONGITUDE,
-    ATTR_ATTRIBUTION, ATTR_LATITUDE, ATTR_LONGITUDE)
+    ATTR_ATTRIBUTION, ATTR_LATITUDE, ATTR_LONGITUDE,
+    LENGTH_KILOMETERS, LENGTH_METERS)
 from homeassistant.helpers.entity import Entity
-from homeassistant.util.location import vincenty
+from homeassistant.util import distance as util_distance
+from homeassistant.util import location as util_location
 import homeassistant.helpers.config_validation as cv
 
 
@@ -65,7 +67,8 @@ class OpenSkySensor(Entity):
         self._session = requests.Session()
         self._latitude = latitude
         self._longitude = longitude
-        self._radius = radius
+        self._radius = util_distance.convert(
+            radius, LENGTH_KILOMETERS, LENGTH_METERS)
         self._state = 0
         self._hass = hass
         self._name = name
@@ -103,10 +106,10 @@ class OpenSkySensor(Entity):
                 continue
             if data.get(ATTR_ON_GROUND):
                 continue
-            distance = vincenty(
-                (self._latitude, self._longitude),
-                (data.get(ATTR_LATITUDE), data.get(ATTR_LONGITUDE)))
-            if distance > self._radius:
+            distance = util_location.distance(
+                self._latitude, self._longitude,
+                data.get(ATTR_LATITUDE), data.get(ATTR_LONGITUDE))
+            if distance is None or distance > self._radius:
                 continue
             callsign = data[ATTR_CALLSIGN].strip()
             if callsign == '':

